src/test/java/net/sf/jabref/exporter/BibtexDatabaseWriterTest.java

```diff
@@ -40,186 +37,175 @@
 
 import com.google.common.base.Charsets;
 import org.junit.Before;
-import org.junit.BeforeClass;
 import org.junit.Test;
 
 import static org.junit.Assert.assertEquals;
-import static org.mockito.Mockito.mock;
 
-public class BibDatabaseWriterTest {
+public class BibtexDatabaseWriterTest {
 
-    private BibDatabaseWriter databaseWriter;
-    private StringWriter stringWriter;
+    private BibtexDatabaseWriter<StringSaveSession> databaseWriter;
     private BibDatabase database;
     private MetaData metaData;
     private BibDatabaseContext bibtexContext;
 
-    @BeforeClass
-    public static void setUpClass() {
+    @Before
+    public void setUp() throws SaveException, IOException {
         Globals.prefs = JabRefPreferences.getInstance();
-    }
 
-    @Before
-    public void setUp() {
-        databaseWriter = new BibDatabaseWriter();
-        stringWriter = new StringWriter();
+        // Write to a string instead of to a file
+        databaseWriter = new BibtexDatabaseWriter<>(StringSaveSession::new);
+
         database = new BibDatabase();
         metaData = new MetaData();
         bibtexContext = new BibDatabaseContext(database, metaData, new Defaults(BibDatabaseMode.BIBTEX));
     }
 
     @Test(expected = NullPointerException.class)
-    public void writeWithNullWriterThrowsException() throws IOException {
-        databaseWriter.writePartOfDatabase(null, bibtexContext, Collections.emptyList(), mock(SavePreferences.class));
-    }
-
-    @Test(expected = NullPointerException.class)
-    public void writeWithNullContextThrowsException() throws IOException {
-        databaseWriter.writePartOfDatabase(mock(Writer.class), null, Collections.emptyList(), new SavePreferences());
+    public void writeWithNullContextThrowsException() throws Exception {
+        databaseWriter.savePartOfDatabase(null, Collections.emptyList(), new SavePreferences());
     }
 
     @Test(expected = NullPointerException.class)
-    public void writeWithNullEntriesThrowsException() throws IOException {
-        databaseWriter.writePartOfDatabase(mock(Writer.class), bibtexContext, null, new SavePreferences());
+    public void writeWithNullEntriesThrowsException() throws Exception {
+        databaseWriter.savePartOfDatabase(bibtexContext, null, new SavePreferences());
     }
 
     @Test(expected = NullPointerException.class)
-    public void writeWithNullPreferencesThrowsException() throws IOException {
-        databaseWriter.writePartOfDatabase(mock(Writer.class), bibtexContext, Collections.emptyList(), null);
+    public void writeWithNullPreferencesThrowsException() throws Exception {
+        databaseWriter.savePartOfDatabase(bibtexContext, Collections.emptyList(), null);
     }
 
     @Test
-    public void writeEncoding() throws IOException {
+    public void writeEncoding() throws Exception {
```

Comment:
Makes the tests more invariant under changes of the code-under-test. (and is default template for IntellliJ)